test/file_image.c

```diff
@@ -780,134 +780,131 @@ test_get_file_image(const char *test_banner, const int file_name_num, hid_t fapl
-
-            size_remaining = image_size;
-            insertion_ptr  = file_image_ptr;
-            i              = 0;
-
-            while (size_remaining > 0) {
-                /* construct the member file name */
-                snprintf(member_file_name, 1024, file_name, i);
-
-                /* open the test file using standard I/O calls */
-                fd = HDopen(member_file_name, O_RDONLY);
-                VERIFY(fd >= 0, "HDopen() failed.");
-
-                if (size_remaining >= FAMILY_SIZE) {
-                    bytes_to_read = FAMILY_SIZE;
-                    size_remaining -= FAMILY_SIZE;
-                }
-                else {
-                    bytes_to_read  = size_remaining;
-                    size_remaining = 0;
-                }
-
-                /* read the member file from disk into the buffer */
-                bytes_read = HDread(fd, insertion_ptr, (size_t)bytes_to_read);
-                VERIFY(bytes_read == bytes_to_read, "HDread() failed.");
-
-                insertion_ptr = (void *)(((char *)insertion_ptr) + bytes_to_read);
-
-                i++;
-
-                /* close the test file */
-                result = HDclose(fd);
-                VERIFY(result == 0, "HDclose() failed.");
-            }
-        }
-        else {
-            /* get the size of the test file */
-            result = HDstat(file_name, &stat_buf);
+    /* Read file from disk */
+    if (is_family_file) {
+        char    member_file_name[1024];
+        ssize_t bytes_to_read;
+        ssize_t member_size;
+        ssize_t size_remaining;
+
+        /*
+         * Modifications need to be made to accommodate userblock when
+         * H5Fget_file_image() works for family driver
+         */
+        i         = 0;
+        file_size = 0;
+
+        do {
+            snprintf(member_file_name, (size_t)1024, file_name, i);
+
+            /* get the size of the member file */
+            result = HDstat(member_file_name, &stat_buf);
             VERIFY(result == 0, "HDstat() failed.");
 
-            /* Since we use the eoa to calculate the image size, the file size
-             * may be larger.  This is OK, as long as (in this specialized instance)
-             * the remainder of the file is all '\0's.
-             */
-            file_size = (ssize_t)stat_buf.st_size;
-            if (user) {
-                VERIFY(file_size > USERBLOCK_SIZE, "file size !> userblock size.");
-                file_size -= USERBLOCK_SIZE;
-            }
+            member_size = (ssize_t)stat_buf.st_size;
+
+            i++;
+            file_size += member_size;
+        } while (member_size > 0);
+
+        /* Since we use the eoa to calculate the image size, the file size
+         * may be larger.  This is OK, as long as (in this specialized instance)
+         * the remainder of the file is all '\0's.
+         */
+        VERIFY(file_size >= image_size, "file size != image size.");
```

Comment:
\>= vs !=?

Reply:
GitHub is matching the lines in the diff incorrectly. This line is unchanged except for indentation.

Reply:
Oh you mean the error message is wrong. I can fix that, though it wasn't changed by this fix.